datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/InstantDeserializer.java

```diff
@@ -277,6 +285,16 @@ protected T _fromString(JsonParser p, DeserializationContext ctxt,
             string = replaceZeroOffsetAsZIfNecessary(string);
         }
 
+        // For some reason DateTimeFormatter.ISO_INSTANT only supports UTC ISO 8601 strings, so it have to be excluded
+        if (_formatter == DateTimeFormatter.ISO_OFFSET_DATE_TIME ||
+            _formatter == DateTimeFormatter.ISO_ZONED_DATE_TIME) {
+
+            // 21-March-2021, Oeystein: Work-around for supporting iso-8601 time offsets that are
```

Comment:
I'd suggest mentioning that this is to support **basic** 8601 format (colonless) vs. **extended** format, to connect the JavaDoc to the 8601 specification.

Reply:
Reworded the comment.